app/Models/Calendar.php

```diff
@@ -812,6 +823,18 @@ public function pastReminders(int $needle = 10): Collection
             ->take($needle)
             ->get();
 
+        foreach ($reminders as $key => $reminder) {
+            if (
+                (!empty($reminder->recurring_until) && empty($reminder->recurring_until_month) && $reminder->recurring_until < $this->currentYear()) ||
+                (!empty($reminder->recurring_until_month) && empty($reminder->recurring_until_day) && $reminder->recurring_until_month < $this->currentMonth() && $reminder->recurring_until < $this->currentYear()) ||
+                (!empty($reminder->recurring_until_day) && ($reminder->recurring_until_day < $this->currentDay() && $reminder->recurring_until_month == $this->currentMonth() && $reminder->recurring_until == $this->currentYear() ||
+                $reminder->recurring_until_month < $this->currentMonth() && $reminder->recurring_until <= $this->currentYear())
```

Comment:
Note for later: I'm confused by the logic here, since it's doing the same logic here compared to in the "upcoming" events loop. Shouldn't it do the reverse? And if it's the same, we should remove the copy-pasting and make it a function